config/m4/java.m4

```diff
@@ -29,8 +29,8 @@ AS_IF([test "x$with_java" != xno],
                             AC_CHECK_PROG(READLINK, readlink, yes)
                             AS_IF([test "x${READLINK}" = xyes],
                                   [
-                                   AC_SUBST([JAVA], [$(readlink -f $(type -P java))])
-                                   AC_SUBST([JAVA_HOME], [${JAVA%*/jre*}])
+                                   AC_SUBST([JAVAC], [$(readlink -f $(type -P javac))])
```

Comment:
better use javac autotools macro
https://www.gnu.org/software/autoconf-archive/ax_prog_java.html#ax_prog_java

assumption that java_home is ../../ away from javac is not valid for many jdks.

Reply:
https://www.baeldung.com/find-java-home
```
dirname $(dirname $(readlink -f $(which javac)))
```
That's basically equivalent of `../../`
Need to find `JAVA_HOME`, it could be in several directories:
https://github.com/OpenNMS/autotools/blob/master/java.m4#L73

Reply:
JDK itself uses `dirname (dirname (javac))`:
http://hg.openjdk.java.net/jdk/jdk/file/d57d61aafef9/make/autoconf/boot-jdk.m4#l132